### PR TITLE
 	Re-activate and Update CNBC.com related ruleset

### PR DESCRIPTION
**Pull request by cschanaj**
`Sublist3r` output more than 400+ sub-domains, only added those which are obvious on `www.cnbc.com`. thanks. 

 - [x] cnbc.com
 - [x] www.cnbc.com
 - [x] apps.cnbc.com
 - [x] data.cnbc.com
 - [x] fm.cnbc.com
 - [x] futuresnow.cnbc.com
 - [x] media.cnbc.com
 - [x] portfolio.cnbc.com
 - [x] pro.cnbc.com
 - [x] sc.cnbc.com
 - [x] video.cnbc.com
 - [x] watchlist.cnbc.com
 - [x] webcast.cnbc.com

Remark: Some videos are broken because of `Demdex.net` and `Dynamic Yield.com` rulesets, disable them manually fix the issues #9641 .

### CODE REVIEW

File: src/chrome/content/rules/CNBC.com.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			 - watchlist.cnbc.com
+
+		Mixed Content Blocking (MCB) tiggered:
+			 - webcast.cnbc.com
+				 - https://webcast.cnbc.com/show/futuresnow
+-->
+<ruleset name="CNBC.com (partial)">
+	<target host="cnbc.com" />
+	<target host="www.cnbc.com" />
+	<exclusion pattern="^http://www\.cnbc\.com/$" />
```
Comment (J0WI) [inline src/chrome/content/rules/CNBC.com.xml]: Could you also provide a test url that works over https?

File: src/chrome/content/rules/CNBC.com.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		<test url="http://mps.cnbc.com/fetch/ext/load-cnbc.com-relaunch.js" />
+	<target host="portfolio.cnbc.com" />
+	<target host="pro.cnbc.com" />
+	<target host="quote.cnbc.com" />
+		<test url="http://quote.cnbc.com/quote-html-webservice/quoteform.htm" />
+	<target host="register.cnbc.com" />
+
+	<!-- Complications -->
+	<rule from="^http://www\.cnbc\.com/(fonts|staticcontent|staticContent)/" 
+			to="https://www.cnbc.com/$1/" />
```
Comment (J0WI) [inline src/chrome/content/rules/CNBC.com.xml]: Most part WFM over https, but with mixed content issues.

**Comment by cschanaj** [comment]
@J0WI this is ready for review again. thanks.

File: src/chrome/content/rules/CNBC.com.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	<target host="quote.cnbc.com" />
+		<test url="http://quote.cnbc.com/quote-html-webservice/quoteform.htm" />
+	<target host="register.cnbc.com" />
+	<target host="pages.response.cnbc.com" />
+
+	<rule from="^http://cnbc\.com/"
+			to="https://cnbc.com/" />
+
+	<rule from="^http://(commentsapi|fm|gdsapi|login|media|mps|portfolio|pro|quote|register|(pages\.response))\.cnbc\.com/"
+			to="https://$1.cnbc.com/" />
```
Comment (J0WI) [inline src/chrome/content/rules/CNBC.com.xml]: That's the same like a simple `<rule from="^http:" to="https:" />`, isn't it?

File: src/chrome/content/rules/CNBC.com-mixed.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<!--
+	For more CNBC.com related, see CNBC.com.xml
+-->
+<ruleset name="CNBC.com (mixed)">
```
Comment (J0WI) [inline src/chrome/content/rules/CNBC.com-mixed.xml]: This rule will break `webcast.cnbc.com`. You should either use the mixedcontent platform or exclusions.

**Comment by cschanaj** [comment]
thanks!!

File: src/chrome/content/rules/CNBC.com-mixed.xml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <!--
 	For more CNBC.com related, see CNBC.com.xml
 -->
-<ruleset name="CNBC.com (mixed)">
+<ruleset name="CNBC.com (mixed)" platform="mixedcontent">
```
Comment (J0WI) [inline src/chrome/content/rules/CNBC.com-mixed.xml]: With the mixedcontent platform, you can remove the exclusions and use a simple redirect.
(or you can move the partial redirect to the main ruleset.)

**Comment by cschanaj** [comment]
@J0WI The `exclusion` has to be there because `https://www.cnbc.com/ -> http://www.cnbc.com/ (301)` but other `www` URLs do not. I have simplified the mixed rule. 

P.S. This `DO NOT` fix #9641 AFAIK.

**Comment by cschanaj** [comment]
Please refer to #9772

**Pull request closed without merging**